Sources/SwiftLexicalLookup/LookupResult.swift

```diff
@@ -20,6 +20,10 @@ import SwiftSyntax
   case fromFileScope(SourceFileSyntax, withNames: [LookupName])
   /// Indicates where to perform member lookup.
   case lookInMembers(LookInMembersScopeSyntax)
+  /// Indicates to lookup generic parameters of extended type.
+  case lookInGenericParametersOfExtendedType(ExtensionDeclSyntax)
+  /// Indicates this closure expression could introduce dollar identifiers.
+  case mightIntroduceDollarIdentifiers(ClosureExprSyntax)
```

Comment:
This enum is turning into pseudo-documentation for all of the weird lookup rules. Could you add a code example for these last two cases? Folks using the lookup APIs are likely to have to handle each of these cases, and a code example will help them understand why they exist and how to test for them.

Reply:
Thanks for the suggestion. I added some example scenarios showing this behavior.

Reply:
Thank you!